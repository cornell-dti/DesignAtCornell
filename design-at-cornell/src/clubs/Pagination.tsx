import React from 'react';
import { StyledPagination, PageButton, PageNavButton } from './ClubsStyles';
import rightArrow from '../static/images/right-arrow.svg';

const Pagination = ({ currentPage, clubsPerPage, totalClubs, paginate }: Props) => {
  const numPages = Math.ceil(totalClubs / clubsPerPage);
  const pageNumbers = [];

  if (numPages > 1) {
    for (let i = 1; i <= numPages; i++) {
      pageNumbers.push(i);
    }
  }

  return (
    <StyledPagination>
      {currentPage > 1 ? (
        <PageNavButton direction="left" onClick={() => paginate(currentPage - 1)}>
          <img src={rightArrow} alt="previous page" />
          Previous
        </PageNavButton>
      ) : (
        <div />
      )}
      <ul>
        {pageNumbers.map((page) => (
          <li key={page}>
            <PageButton selected={page === currentPage} onClick={() => paginate(page)}>
              {page}
            </PageButton>
          </li>
        ))}
      </ul>
      {currentPage < numPages ? (
        <PageNavButton direction="right" onClick={() => paginate(currentPage + 1)}>
          Next
          <img src={rightArrow} alt="next page" />
        </PageNavButton>
      ) : (
        <div />
      )}
    </StyledPagination>
  );
};

export default Pagination;

interface Props {
  currentPage: number;
  clubsPerPage: number;
  totalClubs: number;
  paginate(page: number): void;
}
