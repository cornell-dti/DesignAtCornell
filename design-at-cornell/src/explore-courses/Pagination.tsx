import React from 'react';
import { StyledPagination } from './ExploreCoursesStyles';

const Pagination = ({ currentPage, coursesPerPage, totalCourses, paginate }: Props ) => {
  const numPages = Math.ceil(totalCourses / coursesPerPage);
  const pageNumbers = [];

  if (numPages > 1) {
    for (let i = 1; i <= numPages; i++) {
      pageNumbers.push(i);
    }
  }

  return (
    <StyledPagination>
      <ul>
        {pageNumbers.map(page => (
          <li key={page}>
            <button onClick={() => paginate(page)}>
              {page}
            </button>
          </li>
        ))}
      </ul>
      {(currentPage < numPages)? 
        <button onClick={() => paginate(currentPage + 1)}>
          Next
        </button> : null        
      }
    </StyledPagination>
  );
};

export default Pagination

interface Props {
  currentPage: number
  coursesPerPage: number
  totalCourses: number
  paginate(page: number): void
}