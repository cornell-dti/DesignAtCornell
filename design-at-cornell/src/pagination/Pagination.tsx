import React, { useContext } from 'react';
import { Pagination as SemanticPagination, StrictPaginationProps } from 'semantic-ui-react';
import { StyledPagination } from './PaginationStyles';
import { GlobalContext } from '../context/GlobalContext';

const Pagination = ({ currentPage, cardsPerPage, totalCards, paginate }: Props) => {
  const { isMobileView } = useContext(GlobalContext);

  const numPages = Math.ceil(totalCards / cardsPerPage);

  const updatePage: StrictPaginationProps['onPageChange'] = (e, { activePage }) => {
    paginate(activePage as number);
  };

  return (
    <StyledPagination>
      <SemanticPagination
        pointing
        secondary
        size={isMobileView ? 'tiny' : 'large'}
        firstItem={null}
        lastItem={null}
        activePage={currentPage}
        onPageChange={updatePage}
        totalPages={numPages}
      />
    </StyledPagination>
  );
};

export default Pagination;

interface Props {
  currentPage: number;
  cardsPerPage: number;
  totalCards: number;
  paginate(page: number): void;
}
