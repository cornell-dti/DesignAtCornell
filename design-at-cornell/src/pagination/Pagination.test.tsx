import React from 'react';
import renderer from 'react-test-renderer';
import Pagination from './Pagination';
import 'jest-styled-components';

describe('<Pagination />', () => {
  const props = {
    currentPage: 1,
    cardsPerPage: 12,
    totalCards: 56,
    paginate: () => {},
  };
  it('renders correctly', () => {
    const snapshot = renderer.create(<Pagination {...props} />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('renders correctly with no cards', () => {
    const propsWithNoCourses = { ...props, totalCards: 0 };
    const snapshot = renderer.create(<Pagination {...propsWithNoCourses} />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('renders correctly less cards than coursesPerPage', () => {
    const propsWithFewCourses = { ...props, totalCards: 6 };
    const snapshot = renderer.create(<Pagination {...propsWithFewCourses} />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
