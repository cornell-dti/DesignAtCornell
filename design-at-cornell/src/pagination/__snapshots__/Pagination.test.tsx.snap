// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 80px;
  padding: 0px 80px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 ul {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 70%;
  list-style-type: none;
  padding: 0;
  margin: 0;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 li {
  display: inline-block;
  margin: 10px;
}

.c0 div {
  width: 15%;
  height: 41px;
}

.c1 {
  width: 41px;
  height: 41px;
  border-radius: 100%;
  border: none;
  font-weight: bold;
  font-size: 20px;
  line-height: 23px;
  cursor: pointer;
  color: black;
  background: #E0E0E0;
}

.c1:focus {
  border: none;
  outline: none;
}

.c2 {
  width: 41px;
  height: 41px;
  border-radius: 100%;
  border: none;
  font-weight: bold;
  font-size: 20px;
  line-height: 23px;
  cursor: pointer;
  color: gray;
  background: none;
}

.c2:focus {
  border: none;
  outline: none;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 15%;
  height: 41px;
  border: none;
  background: none;
  font-weight: bold;
  font-size: 20px;
  line-height: 41px;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  cursor: pointer;
}

.c3 img {
  margin: 10px;
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
}

.c3:focus {
  border: none;
  outline: none;
}

<div
  className="c0"
>
  <div />
  <ul>
    <li>
      <button
        className="c1"
        onClick={[Function]}
        selected={true}
      >
        1
      </button>
    </li>
    <li>
      <button
        className="c2"
        onClick={[Function]}
        selected={false}
      >
        2
      </button>
    </li>
    <li>
      <button
        className="c2"
        onClick={[Function]}
        selected={false}
      >
        3
      </button>
    </li>
    <li>
      <button
        className="c2"
        onClick={[Function]}
        selected={false}
      >
        4
      </button>
    </li>
    <li>
      <button
        className="c2"
        onClick={[Function]}
        selected={false}
      >
        5
      </button>
    </li>
  </ul>
  <button
    className="c3"
    direction="right"
    onClick={[Function]}
  >
    Next
    <img
      alt="next page"
      src="right-arrow.svg"
    />
  </button>
</div>
`;

exports[`<Pagination /> renders correctly less cards than coursesPerPage 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 80px;
  padding: 0px 80px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 ul {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 70%;
  list-style-type: none;
  padding: 0;
  margin: 0;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 li {
  display: inline-block;
  margin: 10px;
}

.c0 div {
  width: 15%;
  height: 41px;
}

<div
  className="c0"
>
  <div />
  <ul />
  <div />
</div>
`;

exports[`<Pagination /> renders correctly with no cards 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 80px;
  padding: 0px 80px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 ul {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 70%;
  list-style-type: none;
  padding: 0;
  margin: 0;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 li {
  display: inline-block;
  margin: 10px;
}

.c0 div {
  width: 15%;
  height: 41px;
}

<div
  className="c0"
>
  <div />
  <ul />
  <div />
</div>
`;
